Samples/resource-owner-password-credential/Angular2/Angualar2-Client-ROPC/app/authorize/authorize-component.ts

```diff
@@ -0,0 +1,218 @@
+    }
+    // end
+
+    public Login(creds: logModel) {
+        this.isLoading=true;
+        var instance = this;
+        this.AuthService.Login(creds)
+            .subscribe(
+            Ttoken => {
+                this.logMsg = "You are logged In Now , Please Wait ....";
+                localStorage.setItem("auth_key", Ttoken.access_token);
+                localStorage.setItem("refresh_key", Ttoken.refresh_token);
+                instance.getUserFromServer();
+                this.mclose();
+                this._parentRouter.navigate(['/dashboard']);
+                this.isLoading=false;
+            },
+            Error => {
+                this.isLoading=false;
+                var error, key;
+                key = Error.json();
+                for (error in key) 
+                {
+                    this.logMsg = "";
+                    this.logMsg = this.logMsg + key[error] + "\n";
+                }
+            })
+    }
+
+    public refreshLogin() {
+    this.isLoading=true;
+    var instance = this;
+        this.AuthService.refreshLogin()
+            .subscribe(
+            Ttoken => {
+                this.isLoading=false;
+                this.logMsg = "You are logged In Now , Please Wait ....";
+                localStorage.setItem("auth_key", Ttoken.access_token);
+                localStorage.setItem("refresh_key", Ttoken.refresh_token);
+                instance.getUserFromServer();
+                this.mclose();
+                this._parentRouter.navigate(['/dashboard']);
+            },
+            Error => {
+                this.isLoading=false;
+                var error, key;
+                key = Error.json();
+                for (error in key) 
+                {
+                    this.logMsg = "";
+                    this.logMsg = this.logMsg + key[error] + "\n";
+                }
+            })
+    }
+
+    public getUserFromServer() {
+        this.isLoading=true;
+        var instance = this;
+        this.resourceService.getUserInfo().subscribe(data => {
+            this.isLoading=false;
+            instance.userDetails = data;
+            this.isLoggedin = true;
+            },
+            error => 
+            {
+                this.isLoading=false;
+                instance.userDetails = error;
+            });
+    }
+
+    public Logout() {
+        this.isLoading=true;
+        this.AuthService.logout().subscribe(data => {
+            this.isLoading=false;
+            localStorage.removeItem("auth_key");
+            localStorage.removeItem("refresh_key");
+            this._parentRouter.navigate(['/']);
+            this.isLoggedin = false;
+        }, error => 
+            {
+             this.isLoading=false;
+             this.logMsg = error 
+            });
+    }
+
+    public userRegister(creds:registerModel) {
+        this.isLoading=true;
+        this.AuthService.Register(creds)
+            .subscribe(
+            Ttoken => {
```

Comment:
`Ttoken` is the registration result right? consider renaming it, the name token is forviant